os/pm/pm_register.c

```diff
@@ -100,4 +109,41 @@ int pm_register(FAR struct pm_callback_s *callbacks)
 	return ret;
 }
 
+/****************************************************************************
+ * Name: pm_domain_register
+ *
+ * Description:
+ *   This function is called by a device driver in order to register domain inside PM.
+ *
+ * Input parameters:
+ *   domain - the string domain need to be registered in PM.
+ *
+ * Returned value:
+ *    non-zero integer: ID of domain
+ * 	  -1: On Error
+ *
+ ****************************************************************************/
+
+int pm_domain_register(const char *domain) {
+	int index = 0;
+	int ret = ERROR;
+	/* Iterate over each domain ID and check if given domain is in our domain map or not*/
+	while ((index < CONFIG_PM_NDOMAINS) && pm_domain_map[index]!=NULL) {
+		/* If domain is in our map, then return the corresponding domain ID*/
+		if (strcmp(pm_domain_map[index], domain) == 0) {
+			ret = index;
+			goto errout;
+		}
+		index++;
+	}
+	/* If we have unused domain ID then use it to register given domain */
+	if (index < CONFIG_PM_NDOMAINS) {
+		pm_domain_map[index] = (char *) pm_alloc(UINT8_MAX, sizeof(char));
```

Comment:
UINT8_MAX is long, 
Please add CONFIG_PM_DOMAIN_NAME_SIZE

Reply:
Done